Simplenote/SPNoteEditorViewController+Extensions.swift

```diff
@@ -8,50 +8,58 @@ extension SPNoteEditorViewController {
     ///
     @objc
     var isShowingHistory: Bool {
-        return historyCardViewController != nil
+        return historyViewController != nil
     }
 
     /// Shows note history
     ///
     @objc
     func showHistory() {
         let loader = SPHistoryLoader(note: currentNote)
-        let cardViewController = newHistoryViewController(with: loader, delegate: self)
+        let viewController = newHistoryViewController(with: loader, delegate: self)
 
-        historyCardViewController = cardViewController
+        historyViewController = viewController
         historyLoader = loader
 
-        cardViewController.present(from: self)
+        let transitioningManager = SPCardTransitioningManager()
+        transitioningManager.observer = self
+        historyTransitioningManager = transitioningManager
+
+        present(viewController, with: transitioningManager)
     }
 
-    private func newHistoryViewController(with loader: SPHistoryLoader, delegate: SPNoteHistoryControllerDelegate) -> SPCardViewController {
+    private func newHistoryViewController(with loader: SPHistoryLoader,
+                                          delegate: SPNoteHistoryControllerDelegate) -> UIViewController {
         let controller = SPNoteHistoryController(note: currentNote, loader: loader)
         let historyViewController = SPNoteHistoryViewController(controller: controller)
-        let cardViewController = SPCardViewController(viewController: historyViewController)
 
         controller.delegate = delegate
 
-        return cardViewController
+        return historyViewController
     }
 
     private func dismissHistory() {
-        guard let viewController = historyCardViewController else {
+        guard let viewController = historyViewController else {
             return
         }
 
-        historyCardViewController = nil
-        historyLoader = nil
-
+        cleanUpAfterHistoryDismissal()
         viewController.dismiss(animated: true, completion: nil)
     }
+
+    private func cleanUpAfterHistoryDismissal() {
+        historyViewController = nil
+        historyLoader = nil
+        historyTransitioningManager = nil
```

Comment:
I first did it without nullifying weak vars and what happened is: `save` method in the note editor is checking if the history slider is on screen. If it's on screen, save is skipped. My guess it's done to prevent accidental save happening while user is browsing through the history. To check if history slider is on screen I'm using `historyViewController` var. If I don't nullify it here and user tries to restore to another version of a note, save won't work as the var will still be there (and will be gone only after view controller is fully dismissed).